x-pack/legacy/plugins/infra/public/containers/logs/log_summary/log_summary.test.tsx

```diff
@@ -34,50 +37,43 @@ describe('useLogSummary hook', () => {
   it.skip('queries for new summary buckets when the source id changes', async () => {
     const firstMockResponse = createMockResponse([{ start: 99000, end: 101000, entriesCount: 1 }]);
     const secondMockResponse = createMockResponse([{ start: 99000, end: 101000, entriesCount: 2 }]);
-    const mockApolloClient = {
-      query: jest
-        .fn()
-        .mockResolvedValueOnce(firstMockResponse)
-        .mockResolvedValueOnce(secondMockResponse),
-    };
+
+    fetchLogSummaryMock
+      .mockResolvedValueOnce(firstMockResponse)
+      .mockResolvedValueOnce(secondMockResponse);
 
     const { act, getLastHookValue } = mountHook(
```

Comment:
I think by now we have a working version of `@testing-library/react-hooks`, which can deal with async operations in hooks. That might be a better choice compared to the `mountHook` I wrote before that library was available.

Reply:
I'll give it a shot. If I feel it takes me more than 30 minutes I can create an issue for it and change it afterwards

Reply:
before I commit it, after using the library the test looks like this:

```ts
it('queries for new summary buckets when the source id changes', async () => {
    const firstMockResponse = createMockResponse([{ start: 99000, end: 101000, entriesCount: 1 }]);
    const secondMockResponse = createMockResponse([{ start: 99000, end: 101000, entriesCount: 2 }]);

    fetchLogSummaryMock
      .mockResolvedValueOnce(firstMockResponse)
      .mockResolvedValueOnce(secondMockResponse);

    const { result, waitForNextUpdate, rerender } = renderHook(
      ({ sourceId }) => useLogSummary(sourceId, 100000, 1000, null),
      {
        initialProps: {
          sourceId: 'INITIAL_SOURCE_ID',
        },
      }
    );

    await waitForNextUpdate();
    expect(fetchLogSummaryMock).toHaveBeenCalledTimes(1);
    expect(result.current.buckets).toEqual(firstMockResponse.data.buckets);

    rerender({ sourceId: 'CHANGED_SOURCE_ID' });
    await waitForNextUpdate();

    expect(fetchLogSummaryMock).toHaveBeenCalledTimes(2);
    expect(result.current.buckets).toEqual(secondMockResponse.data.buckets);
  });
```

Reply:
I did rewrite all tests in this file with @testing-library/react-hooks :)